servers/dsub/jobs/test/base_test_cases.py

```diff
@@ -438,3 +468,174 @@ def test_query_jobs_submission_pagination(self):
+            }
+
+            different_labels = {'different-label': 'different-value'}
+
+            labels_no_show = {'no-show': 'cannot-see-me'}
+
+            # Do not change the order of creating jobs since the labels are updated
+
+            # Create a running job that has same label but no test_token label
+            job_running_no_token = self.api_job_id(
+                self.start_job(
+                    'sleep 180',
+                    labels=labels,
+                    name=name,
+                    with_test_token=False))
+            # Create an aborted job
+            job_aborted = self.api_job_id(
+                self.start_job(
+                    'sleep 180',
+                    labels=labels,
+                    name=name,
+                    with_test_token=True))
+            # Create a failed job
+            job_failed = self.api_job_id(
+                self.start_job(
+                    'not_a_command',
+                    labels=labels,
+                    name=name,
+                    with_test_token=True))
+            # Created a succeeded job
+            job_succeeded = self.api_job_id(
+                self.start_job(
+                    'echo SUCCEEDED',
+                    labels=labels,
+                    name=name,
+                    with_test_token=True))
+            # Created a running job
+            job_running = self.api_job_id(
+                self.start_job(
+                    'sleep 180',
+                    labels=labels,
+                    name=name,
+                    with_test_token=True))
+
+            # Create a running job that has a different label and no test_token label
+            job_running_no_token_diff_label = self.api_job_id(
+                self.start_job('sleep 180', labels=labels_no_show, name=name))
+
+            # Create a running job that has different label values
+            job_running_diff_label_value = self.api_job_id(
+                self.start_job(
+                    'sleep 180',
+                    labels=another_value,
+                    name=name,
+                    with_test_token=True))
+            # Create a running job that has different labels
+            job_running_diff_labels = self.api_job_id(
+                self.start_job(
+                    'sleep 180',
+                    labels=different_labels,
+                    name=name,
+                    with_test_token=True))
+
+            # wait_status need 10ms on local provider
+            if self.provider == local.LocalJobProvider(resources):
+                sleep(10)
+
+            self.wait_status(job_aborted, ApiStatus.RUNNING)
+            self.must_abort_job(job_aborted)
+
+            self.wait_status(job_aborted, ApiStatus.ABORTED)
+
+            self.wait_status(job_failed, ApiStatus.FAILED)
+
+            self.wait_status(job_succeeded, ApiStatus.SUCCEEDED)
+
+            self.wait_status(job_running, ApiStatus.RUNNING)
```

Comment:
How about accumulating these running jobs in a list, then waiting for all, then aborting all

Reply:
Not addressed?